### PR TITLE
Request workspace symbols on idle-timeout

### PR DESCRIPTION
**Pull request by the-mikedavis**
Hey sudormrfbin 😀

This is a change on top of #3110 that uses the idle-timeout as a debounce.

The call to highlight the current file feels slightly hacky to me. What do you think?

### CODE REVIEW

**Comment by the-mikedavis** [comment]
cc @sudormrfbin (sorry I don't mean to ping, I just noticed you aren't watching this repo so you may not have seen this. no rush!)

**Comment by sudormrfbin** [comment]
I indeed missed the notification (the title is similar to the PR open in the upstream repo too) so thanks for the ping 😄. I'm away for the weekend though, so it might be a few days till I get to properly test and review the code :)

File: helix-term/src/ui/picker.rs [hunk 1]
```diff
@@ -163,9 +163,15 @@ impl<T: Item> FilePicker<T> {
...
+    }
+
+    pub fn highlight_current_file(&mut self, editor: &Editor) {
         // Try to find a document in the cache
         let doc = self
-            .current_file(cx.editor)
+            .current_file(editor)
             .and_then(|(path, _range)| self.preview_cache.get_mut(&path))
             .and_then(|cache| match cache {
                 CachedPreview::Document(doc) => Some(doc),
```
Comment (sudormrfbin) [inline helix-term/src/ui/picker.rs]: We could use an early return here and drop the Option wrapping

File: helix-term/src/ui/picker.rs [hunk 2]
```diff
@@ -715,28 +721,33 @@ impl<T: Item + Send + 'static> Component for DynamicPicker<T> {
...
-                picker.options = new_options;
-                picker.cursor = 0;
-                picker.force_score();
+                file_picker.picker.options = new_options;
+                file_picker.picker.cursor = 0;
+                file_picker.picker.force_score();
+                // This callback is triggered on the idle timeout, so we simulate the
+                // file-picker's handle_idle_timeout behavior by highlighting the
+                // currently previewed file.
+                file_picker.highlight_current_file(editor);
```
Comment (sudormrfbin) [inline helix-term/src/ui/picker.rs]: Is this needed here? Isn't the highlighting already handled by calling `file_picker.handle_event()` at the beginning of the function, which calls `handle_idle_timeout()`?

https://github.com/helix-editor/helix/blob/26f21da531179ccc77e70b73e32a1b458c4e225b/helix-term/src/ui/picker.rs#L289-L292

**Review by sudormrfbin — COMMENTED** [review]
Sorry for the wait, I'm back! The current approach has the disadvantage of the latency being dependent on a config value that seems unrelated to this -- it might be assumed that the idle timeout config is only used for showing completion popup, so changing it to say 1 second will only update the dynamic picker every second. I suppose we need internal idle timer(s) that are not configurable, but there's no rush since this is seems like a good initial take 👍🏾.

File: helix-term/src/ui/picker.rs (hunk 2 above)
Comment (the-mikedavis) [reply helix-term/src/ui/picker.rs]: If the new symbols response triggered by the idle timeout selects a new file then it won't be highlighted. I just noticed that it doesn't work anyways so I'll remove this 😅

Comment (the-mikedavis) [reply helix-term/src/ui/picker.rs]: (for example searching for something specific like "flip" in the helix repo)

**Comment by sudormrfbin** [comment]
Closing this since upstream PR has been superseded and another one merged; thanks for driving it to completion :)

**Pull request closed without merging**